src/modules/geolocation/components/GeolocationInput.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  Box,
+  Button,
+  FormControl,
+  FormHelperText,
+  Stack,
+  Typography,
+} from '@mui/material';
+
+import React, { useId } from 'react';
+
+import { useGeolocation } from '../hooks/useGeolocation';
+import { ClearIcon, MyLocationIcon } from '@/components/elements/SemanticIcons';
+import { DynamicInputCommonProps } from '@/modules/form/types';
+import BaseMap from '@/modules/geolocation/components/BaseMap';
+import GeolocationErrorAlert from '@/modules/geolocation/components/GeolocationErrorAlert';
+import { LatLon } from '@/modules/geolocation/types';
+
+interface GeolocationInputProps
+  extends Pick<DynamicInputCommonProps, 'label' | 'helperText' | 'disabled'> {
+  value?: LatLon | null;
+  onChange: (value: LatLon | null) => void;
+}
+
+const GeolocationInput: React.FC<GeolocationInputProps> = ({
+  label,
+  value: coordinates,
+  onChange,
+  helperText,
+  disabled,
+}) => {
+  const { requestCoordinates, loading, error } = useGeolocation(onChange);
+  const labelId = useId();
+  const mapHeight = 230;
+
+  return (
+    <>
+      <FormControl sx={{ width: '100%' }}>
+        {label && <div id={labelId}>{label}</div>}
+        {error && <GeolocationErrorAlert error={error} />}
+        <Stack direction='row' sx={{ my: 0.5 }} gap={2} alignItems='center'>
+          <Button
+            onClick={requestCoordinates}
+            startIcon={<MyLocationIcon />}
+            variant='outlined'
+            aria-labelledby={label ? labelId : undefined}
+            disabled={disabled}
+          >
+            Request Location
+          </Button>
+          {coordinates && (
+            <Button
+              variant='gray' // TODO: update to outlined grayscale
+              onClick={() => onChange(null)}
```

Comment:
a11y note - when this button is clicked it causes its own disappearance. If you don't think it's too much code clutter, we can improve the screenreader ux by adding a `buttonRef` to the Request Location button and then calling `buttonRef.current.focus()` here in the `onClick` callback.

Reply:
Thanks, updated! That helps with keyboard nav too.